### PR TITLE
mixin dashbaords: add support for ingest storage replication

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
#### What this PR does

Some panels in the dashboards rely on `cortex_distributor_replication_factor`. This metric is not exported if running in ingest storage (https://github.com/grafana/mimir/pull/7809).

This PR updates all queries in the mixin that were relaying on `cortex_distributor_replication_factor`. If the replication factor metric isn't present now the queries fall back to the maximum of each partition for the metric in question (such as `cortex_ingester_active_series`).

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [x] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: operations/mimir-mixin/dashboards/dashboard-queries.libsonnet [hunk 1]
```diff
@@ -233,5 +234,30 @@ local utils = import 'mixin-utils/utils.libsonnet';
...
+        or on (%(groupByCluster)s)
+        # Ingest storage
+        sum by (%(groupByCluster)s, %(groupByLabels)s) (
+          max by (ingester_id, %(groupByCluster)s, %(groupByLabels)s) (
+            label_replace(
+              %(perIngesterQuery)s,
+              "ingester_id", "$1", "%(instance)s", ".*-([0-9]+)$"
+            )
+          )
+        )
```
Comment (pstibrany) [inline operations/mimir-mixin/dashboards/dashboard-queries.libsonnet:253]: 🤯 Nice :)

File: operations/mimir-mixin/dashboards/writes.libsonnet [hunk 2]
```diff
@@ -61,36 +61,29 @@ local filename = 'mimir-writes.json';
...
-        }, format='short') +
+        $.statPanel(
+          $.queries.ingester.ingestOrClassicDeduplicatedQuery('cortex_ingester_memory_series{%s}' % [$.jobMatcher($._config.job_names.ingester)]),
+          format='short'
+        ) +
         $.panelDescription(
           title,
           |||
             The number of series not yet flushed to object storage that are held in ingester memory.
+            Series are deduplicated according to the replication strategy.
```
Comment (pstibrany) [inline operations/mimir-mixin/dashboards/writes.libsonnet:72]: What is this trying to say? In ingest-storage, there's no replication strategy. (Comment below for ingestion rate "With ingest storage we take the maximum rate of each ingest partition." is more understandable to me)

**Review by pstibrany — APPROVED** [review]
Very nice job!

File: operations/mimir-mixin/dashboards/writes.libsonnet (hunk 2 above)
Comment (dimitarvdimitrov) [reply operations/mimir-mixin/dashboards/writes.libsonnet]: 👍 updated in https://github.com/grafana/mimir/pull/8175/commits/244067fc518f6b142873b934e447c71dcb791385

File: operations/mimir-mixin/dashboards/dashboard-queries.libsonnet [hunk 3]
```diff
@@ -233,5 +234,30 @@ local utils = import 'mixin-utils/utils.libsonnet';
...
       ||| % variables,
     },
+
+    ingester: {
+      ingestOrClassicDeduplicatedQuery(perIngesterQuery, groupByLabels=''):: |||
+        # Classic storage
+        sum by (%(groupByCluster)s, %(groupByLabels)s) (%(perIngesterQuery)s)
+        / on (%(groupByCluster)s) group_left()
+        max by (%(groupByCluster)s) (cortex_distributor_replication_factor{%(distributor)s})
+        or on (%(groupByCluster)s)
```
Comment (pracucci) [inline operations/mimir-mixin/dashboards/dashboard-queries.libsonnet:244]: I don't understand one thing: what is this `on (%(groupByCluster)s)` (in this line) used for?

**Pull request merged**